quiche/src/lib.rs

```diff
@@ -4611,6 +4634,50 @@ impl Connection {
         stream.is_readable()
     }
 
+    /// Returns the next stream that can be written to.
+    ///
+    /// Note that once returned by this method, a stream ID will not be returned
+    /// again until it is "re-armed".
+    ///
+    /// The application will need to fill the stream's buffer and flush the data
+    /// on the wire before the stream is reported again.
+    ///
+    /// This is unlike the [`writable()`] method, that returns the same list of
+    /// writable streams when called multiple times in succession.
+    ///
+    /// [`stream_writable()`]: struct.Connection.html#method.stream_writable
+    /// [`writable()`]: struct.Connection.html#method.writable
+    pub fn stream_writable_next(&mut self) -> Option<u64> {
+        // If there is not enough connection-level send capacity, none of the
+        // streams are writable.
+        if self.tx_cap == 0 {
+            return None;
+        }
+
+        for &stream_id in &self.streams.writable {
+            if let Some(stream) = self.streams.get(stream_id) {
```

Comment:
if the stream isn't in the stream map, should we mark it not writable?

Reply:
The problem is we can't remove elements from `self.streams.writable` while iterating it at the same time.

Reply:
Would it make sense to add a debug_assert then?